### PR TITLE
support protocol v2

### PR DESCRIPTION
**Pull request by BrandonArp**

### CODE REVIEW

File: src/main/java/com/arpnetworking/metrics/common/sources/ClientHttpSourceV2.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2016 Inscope Metrics, Inc.
```
Comment (vjkoskela) [inline src/main/java/com/arpnetworking/metrics/common/sources/ClientHttpSourceV2.java]: 2017

File: src/main/java/com/arpnetworking/metrics/common/sources/ClientHttpSourceV2.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        super(builder);
+    }
+
+    /**
+     * Name of the actor created to receive the HTTP Posts.
+     */
+    public static final String ACTOR_NAME = "appv2";
+
+    /**
+     * ClientHttpSourceV1 {@link BaseSource.Builder} implementation.
```
Comment (vjkoskela) [inline src/main/java/com/arpnetworking/metrics/common/sources/ClientHttpSourceV2.java]: V1 -> V2

File: src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV1ToRecordParser.java [hunk 3]
```diff
@@ -42,7 +42,8 @@
  *
  * @author Brandon Arp (brandon dot arp at inscopemetrics dot com)
  */
-public class ProtobufToRecordParser implements Parser<List<Record>, HttpRequest> {
+@SuppressWarnings("deprecation")
```
Comment (vjkoskela) [inline src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV1ToRecordParser.java]: Please add the javadoc annotation for deprecation as well.

File: src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV2ToRecordParser.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright 2016 Inscope Metrics, Inc.
```
Comment (vjkoskela) [inline src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV2ToRecordParser.java]: 2017

File: src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV1ToRecordParser.java (hunk 3 above)
Comment (vjkoskela) [reply src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV1ToRecordParser.java]: Also should this be a `final` class?

File: src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV2ToRecordParser.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+import java.util.List;
+import java.util.UUID;
+import javax.annotation.Nullable;
+
+/**
+ * Parses the Inscope Metrics protobuf binary protocol into records.
+ *
+ * @author Brandon Arp (brandon dot arp at inscopemetrics dot com)
+ */
+public class ProtobufV2ToRecordParser implements Parser<List<Record>, HttpRequest> {
```
Comment (vjkoskela) [inline src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV2ToRecordParser.java]: Also should this be a `final` class?

File: src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV2ToRecordParser.java [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    private ImmutableMap<String, String> buildDimensions(final ClientV2.Record record) {
+        final ImmutableMap.Builder<String, String> dimensions = ImmutableMap.builder();
+        for (final ClientV2.DimensionEntry dimensionEntry : record.getDimensionsList()) {
+            dimensions.put(dimensionEntry.getName(), dimensionEntry.getValue());
+        }
+        return dimensions.build();
+    }
+}
```
Comment (vjkoskela) [inline src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV2ToRecordParser.java]: Do we not have any tests for v1? We need to add tests for v2.

File: src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV1ToRecordParser.java (hunk 3 above)
Comment (BrandonArp) [reply src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV1ToRecordParser.java]: I didn't deprecate the class, I'm consuming a class that has been deprecated.  In this case, the v1 messages.

File: src/main/java/com/arpnetworking/metrics/common/sources/ClientHttpSourceV2.java (hunk 1 above)
Comment (BrandonArp) [reply src/main/java/com/arpnetworking/metrics/common/sources/ClientHttpSourceV2.java]: Fixed

File: src/main/java/com/arpnetworking/metrics/common/sources/ClientHttpSourceV2.java (hunk 2 above)
Comment (BrandonArp) [reply src/main/java/com/arpnetworking/metrics/common/sources/ClientHttpSourceV2.java]: Fixed

File: src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV1ToRecordParser.java (hunk 3 above)
Comment (BrandonArp) [reply src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV1ToRecordParser.java]: Made final

File: src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV2ToRecordParser.java (hunk 5 above)
Comment (BrandonArp) [reply src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV2ToRecordParser.java]: Fixed

File: src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV2ToRecordParser.java (hunk 4 above)
Comment (BrandonArp) [reply src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV2ToRecordParser.java]: Fixed

File: src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV2ToRecordParser.java (hunk 6 above)
Comment (BrandonArp) [reply src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV2ToRecordParser.java]: Added some tests for you.

**Comment by BrandonArp** [comment]
Added some tests and refactored a bit of the byte[] processing to use ByteString from Akka.  It should cut down on the array copies and reduce memory churn a bit.

File: src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV1ToRecordParser.java (hunk 3 above)
Comment (vjkoskela) [reply src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV1ToRecordParser.java]: Sorry I clearly wasnted thinking on that!

**Comment by vjkoskela** [comment]
Thanks! Looks good to me assuming the rebase build succeeds.

**Pull request merged**